src/fs.cpp

```diff
@@ -12,4 +12,18 @@ FILE *freopen(const fs::path& p, const char *mode, FILE *stream)
     return ::freopen(p.string().c_str(), mode, stream);
 }
 
+// Create and return temporary subdirectory for test outputs
```

Comment:
Small nit: as this is specific to the tests, I'd prefer moving this function to a unit-test specific utilities file, creating one if necessary  -e.g. `src/test/test_utils.cpp`.